File: bhuvigyan/backend/app/api/routes/ndvi.py
from fastapi import APIRouter
from app.models.land import NDVIComputeRequest, NDVITimeseriesRequest
from app.services.ndvi_service import (
    compute_ndvi_for_polygon,
    compute_ndvi_timeseries
)

router = APIRouter()


@router.post("/compute")
async def compute_ndvi(payload: NDVIComputeRequest):
    return await compute_ndvi_for_polygon(
        payload.geojson_geometry,
        payload.survey_number,
        payload.months_back
    )


@router.post("/timeseries")
async def ndvi_timeseries(payload: NDVITimeseriesRequest):
    return await compute_ndvi_timeseries(
        payload.geojson_geometry,
        payload.survey_number,
        payload.months
    )

File: bhuvigyan/frontend/src/api/ndvi.ts
import { apiClient } from './client'

export async function computeNDVI(
  geojson_geometry: any,
  survey_number: string,
  months_back: number = 1
) {
  const { data } = await apiClient.post('/ndvi/compute', {
    geojson_geometry,
    survey_number,
    months_back
  })
  return data
}

export async function computeNDVITimeseries(
  geojson_geometry: any,
  survey_number: string,
  months: number = 12
) {
  const { data } = await apiClient.post('/ndvi/timeseries', {
    geojson_geometry,
    survey_number,
    months
  })
  return data
}
